src/cargo/sources/registry/local.rs

```diff
@@ -34,7 +35,11 @@ impl<'cfg> RegistryData for LocalRegistry<'cfg> {
         &self.index_path
     }
 
-    fn config(&self) -> CargoResult<Option<RegistryConfig>> {
+    fn load(&self, root: &Path, path: &str) -> CargoResult<Vec<u8>> {
```

Comment:
Ah my thinking of passing in `root` is that `index_path` returns a `Filesystem` which is an "unlocked path", but here we've always got a locked path (locked elsewhere) so looking at a `Path` is proof of that.

I was originally unsure what would happen if we take a `\`-separated path when we go down to libgit2, I'm not sure if it handles internally the slash differences. Only one way to find out!